src/shared_utilities/interpolating_time_varying_input1d.jl

```diff
@@ -0,0 +1,165 @@
+import CUDA
+
+"""
+    NearestNeighbor
+
+Return the value corresponding to the point closest to the input time.
+"""
+struct NearestNeighbor <: AbstractInterpolationMethod end
+
+"""
+    LinearInterpolation
+
+Perform linear interpolation between the two neighboring points.
+"""
+struct LinearInterpolation <: AbstractInterpolationMethod end
+
+"""
+    InterpolatingTimeVaryingInput1D
+
+The constructor for InterpolatingTimeVaryingInput1D is not supposed to be used directly, unless you
+know what you are doing. The constructor does not perform any check and does not take care of
+GPU compatibility. It is reasonability of Integrator() to do so.
+
+"""
+struct InterpolatingTimeVaryingInput1D{
+    AA1 <: AbstractArray,
+    AA2 <: AbstractArray,
+    M <: AbstractInterpolationMethod,
+    CC <: ClimaComms.AbstractCommsContext,
+    R <: Tuple,
+} <: AbstractTimeVaryingInput
+    # AA1 and AA2 could be different because of different FTs
```

Comment:
Could we change this to "AA1 and AA2 may have different float types, but they must be the same length, and we assume that they have been sorted to be monotonically increasing in time, without repeated values for the same timestamp." (or we can remove the last clause if we dont make that assumption."

I also wonder if this might go in the docstring instead of a code comment.

Reply:
I think repeated values are fine, but better to be more restrictive. I moved this to the docstring.